fastseq_cli/transformers_generate.py

```diff
@@ -404,10 +440,15 @@ def generate_summaries_or_translations_fast(
                                         BeamSampleEncoderDecoderOutput]):
                         scores_cpu = summaries.sequences_scores.cpu()
                 else: 
-                    scores_cpu = ['NA'] * sequences.shape[0]
+                    scores_cpu = torch.Tensor([float('nan')] * sequences.shape[0])
             if output_summaries_only:
                 sequences = sequences[:, input_ids.shape[-1]:] 
             sequences_cpu = sequences.cpu()
+            if (num_return_sequences is not None and num_return_sequences > 1):
+                sequences_cpu = sequences_cpu.reshape([batch_size, num_return_sequences, sequences_cpu.shape[-1]])
+                if (scores_cpu is not None):
+                    scores_cpu = scores_cpu.reshape([batch_size, num_return_sequences])
+            scores_cpu = np.array(scores_cpu)
```

Comment:
```suggestion
            scores_cpu = scores_cpu.numpy()
```

Could you also add a comment to  explain why `scores_cpu` needs to be converted to numpy?

Reply:
interestingly, this also causes it to hang infinitely. So perhaps the root problem is with multiprocessing and not torch?

Reply:
maybe both are the root causes?

Reply:
Trying to repro it with a smaller example. Will let you know what I find out.